homework_11/homework_11.py

```diff
@@ -0,0 +1,95 @@
+"""
+Написати власний декоратор, задачею якого має бути друк назви функції i часу, коли вона була викликана. Декоратор має працювати для різних функцій однаково.
+
+Написати кастомний Exception клас, MyCustomException, який має повідомляти "Custom exception is occured".
+
+Написати власний менеджер контексту, задачею якого буде друкувати "=========="  10 знаків дорівнює перед виконанням коду та після виконання коду, таким чином виділяючи блок коду символами дорівнює.
+
+Y випадку виникнення будь-якої помилки вона має бути надрукована текстом, проте програма не має завершити своєї роботи.
+
+Написати конструкцію try ... except ... else ... finally, яка буде робити точно те ж, що i менеджер контексту із попереднього завдання.
+"""
+
+#(1)
+
+import time
+
+def decorator(func, *args, **kwargs):
+    def wrapper(*args, **kwargs):
+        print("current time is:", time.strftime('%H'':''%M'))
+        return func()
+    return wrapper
+
+@decorator
+def show():
+    print("name of this function is \"show\"")    
+show()
+
+
+#(3)
+class custom_exception:
+    def __init__(self, value):
+        self.value = value
+        
+    
+    def __enter__(self):
+        print("==========")
+        return True
+    
+    def __exit__(self, type, value, trace):
+        if value == self:
```

Comment:
Here you need to check if there is some error. to make it, you can simply check if there is something in `value`:
```
def __exit__(self, type, value, trace):
    if value:
        print(value)
    print('==========')
    return True